aip/0142.md

```diff
@@ -65,6 +69,28 @@ message is used. For example, a Python-based tool could convert timestamps to
 [datetime][py_datetime] objects and durations to [timedelta][py_timedelta]
 objects.
 
+### Relative timestamps
+
+In some cases, it may be necessary to represent an absolute point in time
+relative to a different start point than the 1970-01-01 typically adopted by
+UNIX timestamps. In these cases, the [`google.protobuf.Timestamp`][timestamp]
+type **should** still be used, however the field comment **must** explain the
+frame of reference in use.
+
+To illustrate this, consider a resource representing a segment of an audio
+stream:
+
+```proto
+message AudioSegment {
+  // The point in time at which the segment starts, relative to the start of
+  // the audio stream.
+  google.protobuf.Timestamp start_time = 1;
```

Comment:
This sounds very wrong to me. If it's relative, it should be a duration. A timestamp is a timestamp - it's not relative to anything. The fact that we use the Unix epoch in Timestamp is a somewhat-leaky implementation detail.

Reply:
I'm pretty sure that a timestamp is absolutely relative to something -- the zero value has to mean something... 

Given the example (of an audio segment), there are two ways to go that I can think of (or remember us doing in the past)

1. Use a timestamp value and say "this is the value relative to the start of the audio stream"
2. Use a duration and call it _offset (e.g., start_offset)

The first is more consistent with a firm start point and a relative duration to mark a span. The second works also, but seems a bit unusual to use a duration to start a specific start-point. Either one is fine with me -- and I'm open to other alternatives that maybe I've overlooked.

Reply:
> I'm pretty sure that a timestamp is absolutely relative to something

Only as an implementation detail. In Noda Time, I use the Unix epoch, but I could change to use a different epoch without that being observable except for binary serialization.

In your audio segment example I would *absolutely* use a duration. You're effectively measuring the time between the start of the stream and the start of the segment. That's a duration, it's *not* a timestamp. We should definitely encourage 2 instead.

(Any time the JSON representation of a resource looks weird, that should be a red flag - and an audio segment that claims to start in 1970 would look very weird to me.)

Reply:
Sounds good -- updating.